sycl/include/sycl/handler.hpp

```diff
@@ -1759,34 +1759,9 @@ class __SYCL_EXPORT handler {
   // TODO: this variant is currently enabled for 2+ reductions only as the
   // versions handling 1 reduction variable are more efficient right now.
   //
-  // Algorithm:
-  // 1) discard_write accessor (DWAcc), InitializeToIdentity = true:
-  //    a) Create uninitialized buffer and read_write accessor (RWAcc).
-  //    b) discard-write partial sums to RWAcc.
-  //    c) Repeat the steps (a) and (b) to get one final sum.
-  //    d) Copy RWAcc to DWAcc.
-  // 2) read_write accessor (RWAcc), InitializeToIdentity = false:
-  //    a) Create new uninitialized buffer (if #work-groups > 1) and RWAcc or
-  //       re-use user's RWAcc (if #work-groups is 1).
-  //    b) discard-write to RWAcc (#WG > 1), or update-write (#WG == 1).
-  //    c) Repeat the steps (a) and (b) to get one final sum.
-  // 3) read_write accessor (RWAcc), InitializeToIdentity = true:
-  //    a) Create new uninitialized buffer (if #work-groups > 1) and RWAcc or
-  //       re-use user's RWAcc (if #work-groups is 1).
-  //    b) discard-write to RWAcc.
-  //    c) Repeat the steps (a) and (b) to get one final sum.
-  // 4) USM pointer, InitializeToIdentity = false:
-  //    a) Create new uninitialized buffer (if #work-groups > 1) and RWAcc or
-  //       re-use user's USM pointer (if #work-groups is 1).
-  //    b) discard-write to RWAcc (#WG > 1) or
-  //       update-write to USM pointer (#WG == 1).
-  //    c) Repeat the steps (a) and (b) to get one final sum.
-  // 5) USM pointer, InitializeToIdentity = true:
-  //    a) Create new uninitialized buffer (if #work-groups > 1) and RWAcc or
-  //       re-use user's USM pointer (if #work-groups is 1).
-  //    b) discard-write to RWAcc (#WG > 1) or
-  //       discard-write to USM pointer (#WG == 1).
-  //    c) Repeat the steps (a) and (b) to get one final sum.
+  // This is basically a tree reduction where we re-use user's reduction
```

Comment:
Can you please (maybe in a separate NFC PR) fix the misprint in the comment 6 lines higher?
"Note thought that the last element" --> "The last element"